libs/agno/agno/models/base.py

```diff
@@ -190,8 +193,16 @@ def _invoke_with_retry(self, **kwargs) -> ModelResponse:
                 else:
                     log_error(f"Model provider error after {self.retries + 1} attempts: {e}")
             except RetryableModelProviderError as e:
+                current_count = kwargs.get("retry_with_guidance_limit", 0)
+                if current_count >= self.retry_with_guidance_limit:
+                    raise ModelProviderError(
+                        message="Max retries with guidance reached", model_name=self.name, model_id=self.id
+                    )
+
+                kwargs.pop("retry_with_guidance", None)
                 kwargs["messages"].append(Message(role="user", content=e.retry_guidance_message, temporary=True))
-                return self._invoke_with_retry(**kwargs, retrying_with_guidance=True)
+                kwargs["retry_with_guidance_limit"] = current_count + 1
```

Comment:
This doesn't make sense? `retry_with_guidance_limit` is a class attribute? Shouldn't you have `retry_with_guidance_count` in the kwargs?